src/xrGame/ui/UIPdaWnd.cpp

```diff
@@ -241,6 +242,15 @@ void CUIPdaWnd::SetActiveSubdialog(const shared_str& section)
         m_pActiveDialog = pUILogsWnd;
     }
 
+    luabind::functor<CUIDialogWndEx*> funct;
+    if (GEnv.ScriptEngine->functor("pda.set_active_subdialog", funct))
+    {
+        CUIDialogWndEx* ret = funct((LPCSTR)section.c_str());
+        CUIWindow* pScriptWnd = ret ? smart_cast<CUIWindow*>(ret) : (0);
+        if (pScriptWnd)
+            m_pActiveDialog = pScriptWnd;
```

Comment:
1. Better code style:
```suggestion
    luabind::functor<CUIDialogWndEx*> functor;
    if (GEnv.ScriptEngine->functor("pda.set_active_subdialog", functor))
    {
        CUIDialogWndEx* scriptWnd = funct(section.c_str());
        CUIWindow* wnd = ret ? smart_cast<CUIWindow*>(scriptWnd) : nullptr;
        if (wnd)
            m_pActiveDialog = wnd;
```
2. Do we even need to use smart cast?

Reply:
1. Updated code and fixed code mistakes in the suggestion
2. I think no, removed it